camus-etl-kafka/src/test/java/com/linkedin/camus/etl/kafka/CamusJobTestWithMock.java

```diff
@@ -193,25 +208,117 @@ private void setupJob1() {
+      assertEquals(msg, e.getMessage());
+    }
+  }
+
+  private void setupJobFailDueToOffsetRangeCallException() {
+    TopicMetadataResponse metadataResponse = mockTopicMetaDataResponse();
+    List<MyMessage> myMessages = messagesWritten.get(TOPIC_1);
+    OffsetResponse offsetResponse = mockOffsetResponse(myMessages);
+    FetchResponse fetchResponse = mockFetchResponse(myMessages);
+    mockConsumerThrowsExceptionForOffsetRangeCall(metadataResponse, offsetResponse, fetchResponse);
+    EasyMock.replay(_mocks.toArray());
+  }
+
+  private void mockConsumerThrowsExceptionForOffsetRangeCall(TopicMetadataResponse metadataResponse,
+      OffsetResponse offsetResponse, FetchResponse fetchResponse) {
+    _simpleConsumer = EasyMock.createNiceMock(SimpleConsumer.class);
+    _mocks.add(_simpleConsumer);
+    EasyMock.expect(_simpleConsumer.send((TopicMetadataRequest)EasyMock.anyObject()))
+      .andReturn(metadataResponse).times(1);
+    EasyMock.expect(_simpleConsumer.getOffsetsBefore((OffsetRequest) EasyMock.anyObject()))
+      .andThrow(new RuntimeException()).anyTimes();
+    EasyMock.expect(_simpleConsumer.clientId()).andReturn(KAFKA_CLIENT_ID).times(1);
+    EasyMock.expect(_simpleConsumer.fetch((FetchRequest)EasyMock.anyObject())).andReturn(fetchResponse).times(1);
+  }
+
+  @Test
+  public void testJobFailDueToOffsetRangeCallError() throws Exception {
+    setupJobFailDueToOffsetRangeCallError();
+    try {
+      job = new CamusJob(props);
+      job.run(TestEtlInputFormat.class, EtlMultiOutputFormat.class);
+      fail("Should have thrown RuntimeException: offset range call failed.");
+    } catch (RuntimeException e) {
+      String msg = "Some topics skipped due to failure in getting latest offset from Kafka leaders.";
+      assertEquals(msg, e.getMessage());
+    }
+  }
+
+  private void setupJobFailDueToOffsetRangeCallError() {
+    TopicMetadataResponse metadataResponse = mockTopicMetaDataResponse();
+    List<MyMessage> myMessages = messagesWritten.get(TOPIC_1);
+    OffsetResponse offsetResponse = mockOffsetResponseWithError(myMessages);
+    FetchResponse fetchResponse = mockFetchResponse(myMessages);
+    mockSimpleConsumer(metadataResponse, offsetResponse, fetchResponse);
+    EasyMock.replay(_mocks.toArray());
+  }
+
+  private OffsetResponse mockOffsetResponseWithError(List<MyMessage> myMessages) {
+    OffsetResponse offsetResponse = EasyMock.createNiceMock(OffsetResponse.class);
+    _mocks.add(offsetResponse);
+    EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{myMessages.size()}).times(1);
+    EasyMock.expect(offsetResponse.offsets(EasyMock.anyString(), EasyMock.anyInt())).andReturn(new long[]{0}).times(1);
+    EasyMock.expect(offsetResponse.hasError()).andReturn(true).anyTimes();
+    return offsetResponse;
+  }
+
+  @Test
+  public void testJobOffsetRangeCallThirdRetrySucceed() throws Exception {
```

Comment:
this test doesn't have verify() or assert(), because it passes if the job succeeds (after three retries).